rust/rope/src/delta.rs

```diff
@@ -145,27 +159,65 @@ impl<N: NodeInfo> Delta<N> {
         Delta { els: els, base_len: base_len }
     }
 
+    /// Produce a summary of the delta. Everything outside the returned interval
+    /// is unchanged, and the old contents of the interval are replaced by new
+    /// contents of the returned length. Equations:
+    ///
+    /// `(iv, new_len) = self.summary()`
+    ///
+    /// `new_s = self.apply(s)`
+    ///
+    /// `new_s = simple_edit(iv, new_s.subseq(iv.start(), iv.start() + new_len), s.len()).apply(s)`
+    pub fn summary(&self) -> (Interval, usize) {
+        let mut els = self.els.as_slice();
+        let mut iv_start = 0;
+        if let Some((&DeltaElement::Copy(0, end), rest)) = els.split_first() {
+            iv_start = end;
+            els = rest;
+        }
+        let mut iv_end = self.base_len;
+        if let Some((&DeltaElement::Copy(beg, end), init)) = els.split_last() {
+            if end == iv_end {
+                iv_end = beg;
+                els = init;
+            }
+        }
+        (Interval::new_closed_open(iv_start, iv_end), Delta::els_len(els))
+    }
+
+    fn els_len(els: &[DeltaElement<N>]) -> usize {
```

Comment:
Doc (and possible rename): this method returns the length of the transformed string.